### PR TITLE
add event_extension var

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
 internal ticket

In tandem with the downstream [PR #44](https://github.com/fivetran/dbt_iterable/pull/44)

**This PR will result in the following new package version:** v0.9.0

Non breaking as the default config for event_extension is still true so existing users won't see any changes. 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes
- Introduces variable `iterable__using_event_extension` to disable the `stg_iterable__event_extension` model. This allows the downstream models to run even if the source `event_extension` table does not exist. For more information on how to configure refer to the [README](https://github.com/fivetran/dbt_iterable_source/blob/main/README.md#enabling-and-disabling-models).

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [x] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Variables tried:

```
    iterable_user_history_pass_through_columns:
      - name: phone_number_line_type
        alias: phone_digits
      - name: additional_properties
    iterable_event_extension_pass_through_columns:
      - name: browser_token
    iterable__using_event_extension: true
#    iterable__using_event_extension: false
    
```
### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

**Comment by fivetran-reneeli** [comment]
will regen docs upon approval

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# dbt_iterable_source v0.8.2
+[PR #31](https://github.com/fivetran/dbt_iterable_source/pull/31) contains the following updates:
+
+## Bug Fixes
+- Introduces variable `iterable__using_event_extension` to disable the `stg_iterable__event_extension` model. This allows the downstream models to run even if the source `event_extension` table does not exist. For more information on how to configure refer to the [README](https://github.com/fivetran/dbt_iterable_source/blob/main/README.md#enabling-and-disabling-models).
```
Comment (fivetran-avinash) [inline CHANGELOG.md:5]:
```suggestion
- Introduces variable `iterable__using_event_extension` to disable the `stg_iterable__event_extension` model. This allows the downstream models to run even if the source `event_extension` table does not exist. For more information on how to configure refer to the [README](https://github.com/fivetran/dbt_iterable_source/blob/main/README.md#step-4-enablingdisabling-models).
```

File: README.md [hunk 2]
```diff
@@ -75,7 +76,7 @@ vars:
 
 ### Passing Through Additional Fields
 
-This package includes fields we judged were standard across Iterable users. However, the Fivetran connector allows for additional columns to be brought through in the `event_extension` and `user_history` objects. Therefore, if you wish to bring them through, leverage our passthrough column variables.
+This package includes fields we judged were standard across Iterable users. However, the Fivetran connector allows for additional columns to be brought through in the `event_extension` and `user_history` objects. Therefore, if you wish to bring them through, leverage our passthrough column variables. For event_extension columns, ensure that `iterable__using_event_extension` is set to True, which is the default.
```
Comment (fivetran-avinash) [inline README.md:79]:
```suggestion
This package includes fields we judged were standard across Iterable users. However, the Fivetran connector allows for additional columns to be brought through in the `event_extension` and `user_history` objects. Therefore, if you wish to bring them through, leverage our passthrough column variables. For `event_extension` columns, ensure that `iterable__using_event_extension` is set to True, which is the default.
```

**Review by fivetran-avinash — COMMENTED** [review]
Thanks @fivetran-reneeli !  Let me know when you've regenerated docs and I can go ahead and approve this.

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-reneeli Approved!

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for working on this PR. This mostly looks good there is just one final update that needs to be applied:
- You need to add the [config: enabled: block](https://github.com/fivetran/dbt_iterable_source/blob/942a45b10db725a21d2e275f986e0754ed00f341/models/src_iterable.yml#L53-L54) to the [src.yml for the event_extension source table](https://github.com/fivetran/dbt_iterable_source/blob/942a45b10db725a21d2e275f986e0754ed00f341/models/src_iterable.yml#L138-L141). This way, if the event_extension model is disabled then the source freshness will not be run for that source. If we do not include this logic the source freshness test will fail.
image

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-reneeli thanks for making these changes. This looks good to me!

**Pull request merged**